files/scripts/arp_update

```diff
@@ -46,5 +46,26 @@ while /bin/true; do
       ip6cmd="ip -6 neigh show | grep -v fe80 | grep $vlan | cut -d ' ' -f 1,3 | $ndisc6cmd"
       eval `eval $ip6cmd`
   done
+
+  # refresh neighbor entries from APP_DB in case of mismatch with kernel
+  DBNEIGH=$(sonic-db-cli APPL_DB keys NEIGH_TABLE*)
+  KERNEIGH4=$(ip -4 neigh show | grep Vlan | cut -d ' ' -f 1,3  --output-delimiter=',')
+  KERNEIGH6=$(ip -6 neigh show | grep -v fe80 | grep Vlan | cut -d ' ' -f 1,3  --output-delimiter=',')
+  for neigh in $DBNEIGH; do
+      intf="$( cut -d ':' -f 2 <<< "$neigh" )"
+      ip="$( cut -d ':' -f 3- <<< "$neigh" )"
+      if [[ $intf == *"Vlan"* ]]; then
+          if [[ $ip == *"."* ]] && [[ ! $KERNEIGH4 =~ "${ip},${intf}" ]]; then
+              pingcmd="timeout 0.2 ping -I $intf -n -q -i 0 -c 1 -W 1 $ip >/dev/null"
```

Comment:
Any reason we use ping instead of arping?  Same thing for ndisc6 vs ping6.

Reply:
No, arping/ndisc6 should work regardless if entries in kernel or not. It will send ARP/NS packets.

Reply:
It doesn't work if kernel entry is not present. Please check logs here. 

```
root@str-7260cx3-acs-7:/# arping -q -w 0 -c 1 -i Vlan1000 192.168.0.30
root@str-7260cx3-acs-7:/# ip -4 neigh show | grep Vlan1000
root@str-7260cx3-acs-7:/# arping -q -w 0 -c 1 -i Vlan1000 192.168.0.30
root@str-7260cx3-acs-7:/# ip -4 neigh show | grep Vlan1000
root@str-7260cx3-acs-7:/# timeout 0.2 ping -I Vlan1000 -n -q -i 0 -c 1 -W 0 192.168.0.30
PING 192.168.0.30 (192.168.0.30) from 192.168.0.1 Vlan1000: 56(84) bytes of data.
root@str-7260cx3-acs-7:/# ip -4 neigh show | grep Vlan1000
192.168.0.30 dev Vlan1000 lladdr 24:8a:07:4c:f5:3a REACHABLE
```

Moreover, even if it *does* work for reachable entries, we want the non-reachable entries to purge by moving to "FAILED" state which can happen only if kernel triggers the ARP.

Reply:
Agreed on the purge case.